crates/crypto/merkle/src/index.rs

```diff
@@ -0,0 +1,120 @@
+/// ``GeneralizedIndex`` is the index of a node in the ``tree``.
+pub(crate) type GeneralizedIndex = u64;
+
+/// Return the given bit of a generalized index.
+/// Note: It is fine to pass ``LeafIndex`` to this function,
+/// as the result will be the same.
+pub(crate) fn get_generalized_index_bit(index: GeneralizedIndex, position: u64) -> bool {
```

Comment:
```suggestion
pub fn get_generalized_index_bit(index: GeneralizedIndex, position: u64) -> bool {
```

same for everywhere else, we don't need to declare `pub(crate)` if we don't want to export these functions outside the crate we can just not make the mod for index.rs public in lib.rs